### PR TITLE
Feature sb 64 vendor/buyer personal page

### PR DESCRIPTION
**Pull request by likeri29**
Created route for updating a user and their profile

The first screen shows the information before the update, and the second screen shows the updated profile and user's name.

The last screenshot shows the info about route in swagger

Screenshot 2024-06-25 at 14 45 32
Screenshot 2024-06-25 at 14 45 51
Screenshot 2024-06-25 at 14 49 36

### CODE REVIEW

File: src/modules/users/users.service.ts [hunk 1]
```diff
@@ -89,4 +90,24 @@ export class UsersService {
 
     return this.usersRepository.findById(userId);
   }
+
+  async updateUserAndProfile(
+    userId: string,
+    updateUerAndProfileDto: UpdateUserAndProfileDto,
```
Comment (VladManukian) [inline src/modules/users/users.service.ts]:
```suggestion
    updateUserAndProfileDto: UpdateUserAndProfileDto,
```

Comment (likeri29) [reply src/modules/users/users.service.ts]: Thanks for noticing it. Fixed

File: src/modules/users/users.controller.ts [hunk 2]
```diff
@@ -16,7 +16,11 @@ import { FileInterceptor } from '@nestjs/platform-express';
...
 import { User } from '~/entities';
-import { UserProfileUpdateDto, UserCardDto } from './dto/index';
+import {
+  UserProfileUpdateDto,
+  UserCardDto,
+  UpdateUserAndProfileDto,
+} from './dto/index';
 import { UploadService } from '../upload/upload.service';
 
 @ApiTags('Users')
```
Comment (lyudad) [inline src/modules/users/users.controller.ts:26]: I assume you need protect controller, no?

Comment (likeri29) [reply src/modules/users/users.controller.ts:26]: All our controllers are protected by default

**Pull request merged**